### PR TITLE
MGMT-8753: download logs for kube-api and hypershift flow

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 1]
```diff
@@ -212,6 +216,76 @@ def download_logs(
...
+        command = (
+            f"oc --kubeconfig {kubeconfig_path} get pods --all-namespaces  "
+            f"-o=custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name   --no-headers "
+            f"--no-headers | xargs -n 2 | cut -d' ' -f1,2 "
+            f"| xargs -n 2 bash -c 'echo getting logs of $1 in $0; "
+            f"oc --kubeconfig {kubeconfig_path} logs $1 -n $0  "
+            f"--all-containers > {output_folder}/logs_$1_hypershift.log'"
+        )
+        log.info("Getting all cluster logs with %s", command)
+        utils.run_command_with_output(command)
```
Comment (eranco74) [inline src/assisted_test_infra/download_logs/download_logs.py:286]: Why do we need this part?
Also, perhaps this should get placed under cluster specific folder ({output_folder}/hypershift/{namespace}/$1.log'?

File: scripts/download_logs.sh [hunk 2]
```diff
@@ -39,6 +40,13 @@ function download_service_logs() {
 }
 
 function download_cluster_logs() {
+
+  if [ ${ENABLE_KUBE_API} == "true" ]; then
+      KUBE_API=true skipper run -i -e JUNIT_REPORT_DIR python3 -m src.assisted_test_infra.download_logs "no_url" ${LOGS_DEST} ${ADDITIONAL_PARAMS}
+      exit 0
```
[the remark this answers is not in the document]
Comment (tsorya) [reply scripts/download_logs.sh:46]: cause i don't want to continue the function

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 1 above)
Comment (tsorya) [reply src/assisted_test_infra/download_logs/download_logs.py:286]: this parts downloads all the logs from hypershift hosted cluster as must gather is not working. If we don't need it , will be great to  remove

File: src/assisted_test_infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py [hunk 3]
```diff
@@ -205,6 +205,33 @@ def _has_required_condition() -> Optional[bool]:
...
+            namespace=namespace,
+        )
+
+        return resources
+
+    @classmethod
+    def list_all_namespaces(
+        cls,
+        crd_api: CustomObjectsApi,
+    ) -> List["ClusterDeployment"]:
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py:227]:
```suggestion
    def list_all_namespaces(cls, crd_api: CustomObjectsApi) -> List["ClusterDeployment"]:
```

File: src/assisted_test_infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py [hunk 4]
```diff
@@ -205,6 +205,33 @@ def _has_required_condition() -> Optional[bool]:
             expected_exceptions=waiting.exceptions.TimeoutExpired,
         )
 
+    @classmethod
+    def list(
+        cls,
+        crd_api: CustomObjectsApi,
+        namespace,
+    ) -> List["ClusterDeployment"]:
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py:213]:
```suggestion
    @classmethod
    def list(cls, crd_api: CustomObjectsApi, namespace) -> List["ClusterDeployment"]:
```

File: src/assisted_test_infra/test_infra/helper_classes/kube_helpers/agent_cluster_install.py [hunk 5]
```diff
@@ -316,3 +316,18 @@ def _get_kubeconfig_secret() -> dict:
 
         with open(kubeconfig_path, "wt") as kubeconfig_file:
             kubeconfig_file.write(b64decode(kubeconfig_data).decode())
+
+    @classmethod
+    def list(
+        cls,
+        crd_api: CustomObjectsApi,
+        namespace,
+    ) -> List["AgentClusterInstall"]:
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/helper_classes/kube_helpers/agent_cluster_install.py:325]:
```suggestion
    @classmethod
    def list(cls, crd_api: CustomObjectsApi, namespace) -> List["AgentClusterInstall"]:
```

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 6]
```diff
@@ -212,6 +215,57 @@ def download_logs(
         log.removeHandler(log_handler)
 
 
+@JunitTestCase()
+def download_logs_kube_api(
```
Comment (eliorerz) [inline src/assisted_test_infra/download_logs/download_logs.py:219]: Maybe try to split this function into 2 or more smaller functions

File: src/assisted_test_infra/download_logs/__main__.py [hunk 7]
```diff
@@ -39,7 +62,7 @@ def main():
         for cluster in clusters:
             if args.download_all or should_download_logs(cluster):
                 download_cluster_logs(
-                    client, cluster, args.dest, args.must_gather, args.update_by_events, pull_secret=args.pull_secret
+                    client, cluster, args.dest, args.must_gather, args.update_by_events, args.pull_secret
```
Comment (eliorerz) [inline src/assisted_test_infra/download_logs/__main__.py:65]: That will pass `pull_secret` as `retry_interval` argument

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: src/assisted_test_infra/download_logs/__main__.py (hunk 7 above)
Comment (tsorya) [reply src/assisted_test_infra/download_logs/__main__.py:65]: mistake sorry

File: src/assisted_test_infra/test_infra/helper_classes/kube_helpers/agent_cluster_install.py (hunk 5 above)
Comment (tsorya) [reply src/assisted_test_infra/test_infra/helper_classes/kube_helpers/agent_cluster_install.py:325]: please decide on the style in all the functions in that case as i don't understand styling in that case

File: src/assisted_test_infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py (hunk 4 above)
Comment (tsorya) [reply src/assisted_test_infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py:213]: same as in previous

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 6 above)
Comment (tsorya) [reply src/assisted_test_infra/download_logs/download_logs.py:219]: will move logic of must-gather for kube-api

**Pull request merged**